File: 2017/d2/sol1.ex
File.read!("input.txt")
  |> String.trim
  |> String.split("\n", trim: true)
  |> Enum.reduce(0, fn(x, acc) ->
      {min, max} = String.split(x)
        |> Enum.min_max_by(fn(x) -> String.to_integer(x) end)
      acc + (String.to_integer(max) - String.to_integer(min))
    end)
  |> IO.puts

File: 2017/d7/sol1.rb
input = File.read("input.txt").split("\n")

bases = []
tops = []

input.each do |line|
  parts = line.split(/,?\s+/)
  if parts.length > 2
    bases.push parts[0]
    tops = tops + parts[3..-1]
  end
end

puts bases.length
puts tops.length
puts bases - tops